docs/howto_guides/Realtime-Best-Practices.md

```diff
@@ -0,0 +1,497 @@
+## Do's, Don'ts, and Wisdom
+
+---
+### 1 Trick
+---
+
+#### 1.1 Trick events are computationally expensive. Use them judiciously.
+
+Trick events can provide a quick, easy, way to customize the behavior of a sim, based on some condition. But, because they require Python interpretation, they are slow. They are not intended for implementation of permanent sim functionality. If they are over used, they can seriously degrade simulation performance. So, take it easy with the events.
+
+See [Event Manager] (https://nasa.github.io/trick/documentation/simulation_capabilities/Event-Manager).
+
+#### 1.2 Disable Trick run-time components that your sim doesn't need.
+
+```default_trick_sys.sm```, the file included at the top of any Trick ```S_define``` file defines numerous "modules" that provide functionality to a Trick sim. Whereas some of these modules ( like the Executive, MemoryManager, CommandLineArguments) are required for any Trick Simulation to function, many are optional. If the modules are not needed, then disabling them can improve simulation perfromance.
+
+Inserting one or more of the ```#define``` statements listed below to the top of the ```S_define```, just before  the inclusion of ```default_trick_sys.sm``` will disable those modules.
+
+```
+#define TRICK_NO_EXECUTIVE
+#define TRICK_NO_MONTE_CARLO
+#define TRICK_NO_MEMORY_MANAGER
+#define TRICK_NO_CHECKPOINT_RESTART
+#define TRICK_NO_SIE
+#define TRICK_NO_COMMANDLINEARGUMENTS
+#define TRICK_NO_MESSAGE
+#define TRICK_NO_INPUTPROCESSOR
+#define TRICK_NO_VARIABLE_SERVER
+#define TRICK_NO_DATA_RECORD
+#define TRICK_NO_REALTIME
+#define TRICK_NO_FRAMELOG
+#define TRICK_NO_MASTERSLAVE
+#define TRICK_NO_INSTRUMENTATION
+#define TRICK_NO_INTEGRATE
+#define TRICK_NO_REALTIMEINJECTOR
+#define TRICK_NO_ZEROCONF
+#define TRICK_NO_UNITTEST
+```
+##### Example
+Using "SIM_submarine" as an example, the following demonstrates disabling unneeded ```default_trick_sys.sm``` modules.
+
+```
+/************************************************************
+PURPOSE:
+    ( Simulate a submarine. )
+LIBRARY DEPENDENCIES:
+    ((submarine/src/Submarine.cpp))
+*************************************************************/
+#define TRICK_NO_MONTE_CARLO
+#define TRICK_NO_MASTERSLAVE
+#define TRICK_NO_INSTRUMENTATION
+#define TRICK_NO_REALTIMEINJECTOR
+#define TRICK_NO_ZEROCONF
+#define TRICK_NO_UNITTEST
+#include "sim_objects/default_trick_sys.sm"
+
+##include "submarine/include/Submarine.hh"
+
+class SubmarineSimObject : public Trick::SimObject {
+    public:
+    
+    ...
+```
+
+#### 1.3 Consider running Trick variable server clients and Sims on different machines.
+Trick variable server clients communicate with a simulation via a TCP/IP connection.
+The client process may, but isn't required to run on the same machine as your simulation process. On the same machine, both will competing for the same resources. This can degrade sim performance, especially when clients are rendering high-definition graphics.

```

Comment:
may, but isn't required to -> may, but isn't required to,

Reply:
done.